src/modules/mysql.nix

```diff
@@ -47,6 +47,18 @@ let
       fi
     '') cfg.initialDatabases}
 
+    ${optionalString (cfg.rootPassword != null) ''
+      # Set root password
+
+      if [[ ! -f "$MYSQL_HOME/.root_password_set" ]]; then
+        ${cfg.package}/bin/mysql ${mysqlOptions} -u root -N -e "ALTER USER 'root'@'localhost' IDENTIFIED BY '${cfg.rootPassword}';"
+
+        touch "$MYSQL_HOME/.root_password_set"
```

Comment:
This should include the password hash so if it changes, the password is changes

Reply:
You need the old password to set the new one

Reply:
Maybe we allow adding additional users using options and let the `root` be root and only accessible using socket?